apps/src/lib/kits/maker/toolkit.js

```diff
@@ -0,0 +1,118 @@
+/*
+ * @file Static interface to Maker Toolkit, to simplify App Lab code interfacing
+ * with maker and provide clean setup/cancel/reset patterns.
+ */
+import codegen from '../../../codegen';
+import {getStore} from '../../../redux';
+import CircuitPlaygroundBoard from './CircuitPlaygroundBoard';
+import * as commands from './commands';
+import * as dropletConfig from './dropletConfig';
+import MakerError, {ConnectionCanceledError} from './MakerError';
+import {findPortWithViableDevice} from './portScanning';
+import * as redux from './redux';
+
+// Re-export some modules so consumers only need this 'toolkit' module
+export {dropletConfig, MakerError};
+
+/**
+ * @type {CircuitPlaygroundBoard} The current board controller, populated when
+ * connected, null when not connected.  There can be only one at any time.
+ */
+let currentBoard = null;
+
+/**
+ * Enable Maker Toolkit for the current level.
+ */
+export function enable() {
+  getStore().dispatch(redux.enable());
+}
+
+/**
+ * @returns {boolean} whether Maker Toolkit is enabled for the current level
+ */
+export function isEnabled() {
+  return redux.isEnabled(getStore().getState());
+}
+
+/**
+ * Called when starting execution of the student app.
+ * Looks for a connected board, sets up an appropriate board controller,
+ * and injects needed references into the interpreter and its commands.
+ * @param {JSInterpreter} interpreter
+ * @param {function} onDisconnect
+ * @return {Promise}
+ *   Resolves (with no value) when connection is successful and the board
+ *   controller is ready to use.
+ *   Resolves immediately if maker toolkit is disabled.
+ *   Rejects with a MakerError when the connection process is cancelled or
+ *   fails in an expected way that we handle gracefully (for example, no board
+ *   plugged in).
+ *   Rejects with a plain Error if something more unexpected happens.
+ */
+export function connect({interpreter, onDisconnect}) {
```

Comment:
Documentation seems good. 

Nit: it's a little surprising that it "Resolves immediately if maker toolkit is disabled", but this seems fine given that you've documented it clearly. Could consider calling it `connectIfEnabled`, or not calling `connect` if maker is not enabled, but what you have here is also fine.